src/main/java/seedu/address/logic/parser/EditGenericCommandParser.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_OPTION_PATIENT_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_OPTION_TASK_INDEX;
+
+import seedu.address.logic.commands.EditGenericCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates either a new EditPatientCommand object or
+ * a new EditTaskCommand object based on option values.
+ */
+public class EditGenericCommandParser implements Parser<EditGenericCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditGenericCommand
+     * and returns an EditGenericCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditGenericCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        ArgumentMultimap options = ParserUtil.parseOptions(args, PREFIX_OPTION_PATIENT_INDEX, PREFIX_OPTION_TASK_INDEX);
+        args = ParserUtil.eraseOptions(args, PREFIX_OPTION_PATIENT_INDEX, PREFIX_OPTION_TASK_INDEX);
+
+        if (options.getValue(PREFIX_OPTION_PATIENT_INDEX).isPresent()
+                && !options.getValue(PREFIX_OPTION_TASK_INDEX).isPresent()) {
```

Comment:
A code quality suggestion: would it be better to evaluate `options.getValue(PREFIX_OPTION_PATIENT_INDEX).isPresent()` and  `options.getValue(PREFIX_OPTION_TASK_INDEX).isPresent()` first as an intermediate value so the condition doesn't look too long? Perhaps just `options.getValue(PREFIX_OPTION_PATIENT_INDEX)` and `options.getValue(PREFIX_OPTION_TASK_INDEX)` since they are used multiple times with `get()` too?

Reply:
Done